### DISCUSSION TITLE
Add support for Azure Container Apps and DAPR.io

### DISCUSSION THREAD

**Discussion by MedAnd** [Azure/data-api-builder#1331] [Ideas]
Hi @yorek, great to see MS adopting more support for GraphQL, which is especially prevalent in the retail technology landscape. 

Please consider adding support for Azure Container Apps and DAPR.io

**Comment by aaronpowell** [comment]
What sort of support would you be looking for those?

As DAB is available as a container image, you can deploy that to any of our container hosting options, AppService, Container Instances, Container Apps or even AKS.

We have this outlined in the [`/samples/azure`](https://github.com/Azure/data-api-builder/blob/main/samples/azure/readme.md) for Container Instances, but the pattern would be the same for Container Apps (primarily it'd be changing the some of the CLI commands from the provided script to use `az containerapp`).

For Dapr, I only really have a surface level understanding of how to use it, could you provide an example of what you'd be looking for in Dapr integration?

**Comment by MedAnd** [comment]
Hi @aaronpowell - maybe all that is needed is a GitHub example and docs, which show step by step how to host within Container Apps, and also call / use from other DAPR containers. Wondering if there is any special configuration required to have DAPR and GraphQL play nice?

**Reply by aaronpowell** [reply]
Yes, a sample that shows Container Apps wouldn't be a difficult task, it's a case of porting the existing Azure sample to a different container host.

From memory Dapr can make HTTP calls between services, and assuming that I'm right about that, you could call it just like any other service couldn't you?

**Comment by MedAnd** [comment]
Wondering if a Dapr native GraphQL binding would be an option... so one approach is to simply host within Container Apps, but another is to deploy the data-api-builder along with the Dapr side car and expose the data-api-builder through Dapr, if that made sense?

[Dapr GraphQL binding spec](https://docs.dapr.io/reference/components-reference/supported-bindings/graghql/)

**Reply by aaronpowell** [reply]
Having a glance over that spec, I don't see why it wouldn't be possible to do that, or you could use DAB's REST endpoints through Dapr's HTTP bindings - https://docs.dapr.io/reference/components-reference/supported-bindings/http/

**Reply by MedAnd** [reply]
If the example and docs could contain this scenario, I think this would be a win win win, for Container Apps / Dapr and data-api-builder adoption and end customers... I see great potential in combining, maybe packaging as part of modern Data Mesh services 🙂
